connectivity/FEATURE_BLE/cordio/TARGET_CORDIO/source/PalGapImpl.cpp

```diff
@@ -323,13 +325,29 @@ ble_error_t PalGap::create_connection(
 }
 
 
-ble_error_t PalGap::cancel_connection_creation()
-{
-    DmConnClose(
-        DM_CLIENT_ID_APP,
-        /* connection handle - invalid */ DM_CONN_ID_NONE,
-        /* reason - invalid (use success) */ 0x00
-    );
+ble_error_t PalGap::cancel_connection_creation(
+    peer_address_type_t peerAddressType,
+    const ble::address_t &peerAddress
+) {
+    uint8_t connection_id;
+
+    for (connection_id = 0; connection_id < DM_CONN_MAX; connection_id++) {
+        if (dmConnCb.ccb[connection_id].inUse &&
+            dmConnCb.ccb[connection_id].state == DM_CONN_SM_ST_CONNECTING &&
+            BdaCmp(dmConnCb.ccb[connection_id].peerAddr, peerAddress.data())) {
+            break;
+        }
+    }
+
+    if (connection_id < DM_CONN_MAX) {
+        connection_id++;
+        DmConnClose(
+            DM_CLIENT_ID_APP,
+            /* connection handle */ connection_id,
+            /* reason - invalid (use success) */ 0x00
+        );
+    }
+
     return BLE_ERROR_NONE;
```

Comment:
The `DmConnClose` call can be nested into the for loop so we don't need to check `DM_CONN_MAX` twice. In the loop, return `BLE_ERROR_NONE` after `DmConnClose`. If we reach the end, no address matches so we could return some kind of error.

Reply:
thanks, changed as requested